test/cluster/lwt/test_lwt_with_counters_during_tablets_resize_and_migrations.py

```diff
@@ -0,0 +1,419 @@
+                    attempt,
+                )
+                continue
+
+            dst_server = random.choice(same_rack_candidates)
+            dst_host_id = server_id_to_host_id[dst_server.server_id]
+            dst_shard = 0
+
+        try:
+            await _move_tablet_with_retry(
+                manager, src_server, tester.ks, tester.tbl,
+                src_host_id, src_shard, dst_host_id, dst_shard, token,
+                timeout_s=60
+            )
+            await _move_tablet_with_retry(
+                manager, src_server, tester.ks, counter_table,
+                src_host_id, src_shard, dst_host_id, dst_shard, token,
+                timeout_s=60
+            )
+
+            migration_count += 1
+            logger.info(
+                "Completed migration #%d (token=%s -> %s:%d) for %s.%s and %s.%s",
+                migration_count, token, dst_server.ip_addr, dst_shard,
+                tester.ks, tester.tbl, tester.ks, counter_table,
+            )
+            await asyncio.sleep(random.uniform(*pause_range))
+            continue
+        except Exception as e:
+            if _is_tablet_in_transition_http_error(e):
+                logger.info("Token %s in transition, switching token (attempt %d)",
+                            token, attempt)
+                continue
+            if _is_no_replica_on_src_error(e) or _is_dst_already_replica_error(e):
+                logger.info("Src replica vanished for token %s, re-pick (attempt %d)",
+                            token, attempt)
+                continue
+            raise
+
+    assert migration_count == num_ops, f"Only completed {migration_count}/{num_ops} migrations"
+    logger.info("Completed tablet migration ops: %d/%d", migration_count, num_ops)
+
+
+def powers_of_two_in_range(lo: int, hi: int):
+    if lo > hi or hi < 1:
+        return []
+    lo = max(1, lo)
+    start_e = (lo - 1).bit_length()
+    end_e = hi.bit_length()
+    return [1 << e for e in range(start_e, end_e + 1) if (1 << e) <= hi]
+
+
+async def run_random_resizes(
+    stop_event_: asyncio.Event,
+    manager: ManagerClient,
+    servers,
+    tester: BaseLWTTester,
+    ks: str,
+    table: str,
+    counter_table: str,
+    target_steps: int = TARGET_RESIZE_COUNT,
+    pause_range=(0.5, 2.0),
+):
+    """
+    Perform randomized tablet count changes (splits/merges) on the main LWT table
+    and its counter table. Runs until target resize count is reached or stop_event_
+    is set. Returns a dict with simple stats.
+    """
+    split_count = 0
+    merge_count = 0
+    current_resize_count = 0
+    pow2_targets = powers_of_two_in_range(MIN_TABLETS, MAX_TABLETS)
+
+    while not stop_event_.is_set() and current_resize_count < target_steps:
+        # Drive resize direction from the main table.
+        current_main = await get_tablet_count(manager, servers[0], ks, table)
+        current_counter = await get_tablet_count(manager, servers[0], ks, counter_table)
```

Comment:
nit: this variable seems redundant -- both lwt and counters have the same number of tablets